src/Distributions/Distributions.jl

```diff
@@ -0,0 +1,378 @@
+"""
+  particle mass distribution module
+
+Particle mass distribution functions for microphysical process modeling:
+  - computation of moments of distributions
+  - computation of densities of distributions
+  - creating distributions given a set of parameters
+  - creating distributions given a set of moments
+"""
+module Distributions
+
+using SpecialFunctions: gamma, gamma_inc
+
+import LinearAlgebra: norm
+import Optim: optimize, LBFGS
+
+# mass distributions available for microphysics
+export Distribution
+export Primitive
+export Exponential
+export Gamma
+export Mixture
+
+# methods that query particle mass distributions
+export moment
+export density
+
+# setters and getters
+export update_params_from_moments
+
+
+"""
+  Distribution{FT}
+
+A particle mass distribution function, which can be initialized
+for various subtypes of assumed shapes in the microphysics parameterization.
+"""
+abstract type Distribution{FT} end
+
+
+"""
+  Primitive{FT}
+
+A particle mass distribution that has support on the positive real 
+axis and analytic expressions for moments and partial moments.
+"""
+abstract type Primitive{FT} <: Distribution{FT} end
+
+
+"""
+  Exponential{FT} <: Primitive{FT}
+
+Represents particle mass distribution function of exponential shape.
+
+# Constructors
+  Exponential(n::Real, θ::Real)
+
+# Fields
+
+"""
+
```

Comment:
I think there can't be any free line between the docstring and the actual struct or function.